src/features/steps/pdm-dataset-generation.py

```diff
@@ -150,3 +153,48 @@ def metadata_table_step_impl(context):
     assert (
         item["S3_Prefix_Analytical_DataSet"]["S"] == context.pdm_test_input_s3_prefix
     ), f"S3_Prefix_Analytical_DataSet was '{item['S3_Prefix_Analytical_DataSet']['S']}', expected '{context.pdm_test_input_s3_prefix}'"
+
+
+@then("the pdm_object_tagger has run successfully")
+def step_impl(context):
+    job_id = aws_helper.trigger_batch_job(
+        job_queue_name="pdm_object_tagger",
+        job_name="object-tagging_pdm-object-tagging-e2e",
+        job_definition="s3_object_tagger_job",
+        parameters={
+            "data-s3-prefix": context.pdm_data_prefix,
+            "csv-location": os.path.join(
+                "s3://",
+                context.common_config_bucket,
+                context.pdm_data_classfication_csv,
+            ),
+        },
+    )
+    batch_job_status = aws_helper.poll_batch_job_status(job_id=job_id)
+    assert batch_job_status == "SUCCEEDED"
+
+
+@then("the correct tags are applied to the pdm data")
+def step_impl(context):
+    common_config_bucket = context.common_config_bucket
+    rbac_csv_s3_key = context.pdm_data_classfication_csv
+    pdm_output_prefix = context.pdm_data_prefix
+    published_bucket = context.published_bucket
+
+    all_rbac_tags = object_tagger_helper.get_rbac_csv_tags(
+        rbac_csv_s3_bucket=common_config_bucket, rbac_csv_s3_key=rbac_csv_s3_key
+    )
+
+    s3_keys = aws_helper.get_s3_file_object_keys_matching_pattern(
+        s3_bucket=published_bucket,
+        s3_prefix=pdm_output_prefix,
+    )
+
+    _ = list(
+        map(
+            object_tagger_helper.verify_s3_object_required_tags,
+            itertools.repeat(published_bucket),
+            s3_keys[:50],
```

Comment:
Is there a scenario where we maybe checking the same keys over and over again? If so, should we take into consideration the last_modified date? If not, then ignore this comment.